runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java

```diff
@@ -50,75 +43,65 @@ public void addAllMaps(Collection<T> maps) {
         }
     }
 
-    /** Checkpoint multiple SMRMaps serially.
-     *
-     * @param rt CorfuRuntime
-     * @param author Author's name, stored in checkpoint metadata
-     * @return Global log address of the first record of
-     */
-    public Token appendCheckpoints(CorfuRuntime rt, String author)
-            throws Exception {
-        return appendCheckpoints(rt, author, (x,y) -> { });
-    }
 
-    /** Checkpoint multiple SMRMaps serially.
+    /** Checkpoint multiple SMRMaps. Since this method is Map specific
+     *  then the keys are unique and the order doesn't matter.
      *
      * @param rt CorfuRuntime
      * @param author Author's name, stored in checkpoint metadata
-     * @param postAppendFunc User-supplied lambda for post-append action on each
-     *                       checkpoint entry type.
      * @return Global log address of the first record of
      */
 
-    public Token appendCheckpoints(CorfuRuntime rt, String author,
-                                  BiConsumer<CheckpointEntry, Long> postAppendFunc) {
-
-        rt.getObjectsView().TXBuild()
-                .type(TransactionType.SNAPSHOT)
-                .build()
-                .begin();
-
-        Token snapshot = TransactionalContext.getCurrentContext().getSnapshotTimestamp();
+    public Token appendCheckpoints(CorfuRuntime rt, String author) {
+        log.info("appendCheckpoints: appending checkpoints for {} maps", maps.size());
 
-        log.trace("appendCheckpoints: author '{}' at globalAddress {} begins",
-                author, snapshot);
+        // TODO(Maithem) should we throw an exception if a new min is not discovered
+        Token minSnapshot = Token.UNINITIALIZED;
 
-        log.info("appendCheckpoints: appending checkpoints for {} maps", maps.size());
         final long cpStart = System.currentTimeMillis();
         try {
             for (ICorfuSMR<T> map : maps) {
                 UUID streamId = map.getCorfuStreamID();
                 final long mapCpStart = System.currentTimeMillis();
                 int mapSize = ((T) map).size();
                 CheckpointWriter<T> cpw = new CheckpointWriter(rt, streamId, author, (T) map);
-                cpw.setEnablePutAll(enablePutAll);
                 ISerializer serializer =
                         ((CorfuCompileProxy<Map>) map.getCorfuSMRProxy())
                                 .getSerializer();
                 cpw.setSerializer(serializer);
-                cpw.setPostAppendFunc(postAppendFunc);
                 log.trace("appendCheckpoints: checkpoint map {} begin",
                         Utils.toReadableId(map.getCorfuStreamID()));
-                List<Long> addresses = cpw.appendCheckpoint();
+                Token minCPSnapshot = cpw.appendCheckpoint();
+
+                if (minSnapshot == Token.UNINITIALIZED) {
```

Comment:
It would always come in here, right? we've never changed it since initialized to this value.